pallets/avn/src/tests/tests.rs

```diff
@@ -1,64 +1,91 @@
-use crate::mock::*;
+use crate::{mock::*, OperationType};
 use sp_runtime::testing::UintAuthorityId;
 
 #[test]
-fn test_is_primary_blocknumber_1() {
+fn next_validator_is_incremented_correctly_for_ethereum_operations() {
     let mut ext = ExtBuilder::build_default().with_validators().as_externality();
     ext.execute_with(|| {
-        let block_number = 1;
-        let expected_primary = 2;
-        let result = AVN::is_primary(block_number, &expected_primary);
-        assert!(result.is_ok(), "Getting primary validator failed");
-        assert_eq!(result.unwrap(), true);
+        let eth_index_before: u8 = AVN::get_primary_validator().0;
+        AVN::calculate_primary_validator(OperationType::Ethereum);
+        let eth_index_after: u8 = AVN::get_primary_validator().0;
+        assert_eq!(eth_index_after, eth_index_before + 1);
     });
 }
 
 #[test]
-fn test_is_primary_blocknumber_2() {
+fn the_first_validator_is_picked_again_after_the_last_validator_for_ethereum_operations() {
     let mut ext = ExtBuilder::build_default().with_validators().as_externality();
     ext.execute_with(|| {
-        let block_number = 2;
-        let expected_primary = 3;
-        let result = AVN::is_primary(block_number, &expected_primary);
-        assert!(result.is_ok(), "Getting primary validator failed");
-        assert_eq!(result.unwrap(), true);
+        let validators = AVN::validators();
+        let eth_index_before: u8 = AVN::get_primary_validator().0;
```

Comment:
I'd please ask for this to be replaced by a single function somewhere, in all the tests, that retrieves the right index. For AVN and for Ethereum.
This is too low level, and if the actual implementation changes, there is a lot in the tests that will have to change as well. Even better, can we use a function from the actual code, if it already exists?